.eslintrc.js

```diff
@@ -0,0 +1,229 @@
+  extends: [
+    "eslint:recommended",
+    "plugin:react/recommended",
+    "plugin:react/jsx-runtime",
+    "plugin:react-hooks/recommended",
+    "plugin:import/errors",
+    "plugin:import/warnings",
+    "plugin:import/typescript",
+    ...(shouldLintCssModules ? ["plugin:postcss-modules/recommended"] : []),
+  ],
+  settings: {
+    "import/internal-regex": "^metabase/|^metabase-lib/",
+    "import/resolver": {
+      webpack: {
+        typescript: true,
+      },
+    },
+    "import/ignore": ["\\.css$"],
+    react: {
+      version: "detect",
+    },
+    "postcss-modules": {
+      baseDir: "./frontend/src",
+    },
+  },
+  parserOptions: {
+    ecmaFeatures: {
+      legacyDecorators: true,
+    },
+  },
+  overrides: [
+    {
+      files: ["*.js", "*.jsx", "*.ts", "*.tsx"],
+      rules: {
+        "no-unconditional-metabase-links-render": "error",
+        "no-literal-metabase-strings": "error",
+      },
+    },
+    {
+      files: [
+        "*.unit.spec.*",
+        "frontend/src/metabase/admin/**/*",
+        "frontend/src/metabase/setup/**/*",
+        "frontend/lint/**/*",
+        "*.stories.*",
+        "e2e/**/*",
+        "**/tests/*",
+        "release/**/*",
+      ],
+      rules: {
+        "no-unconditional-metabase-links-render": "off",
+        "no-literal-metabase-strings": "off",
+      },
+    },
+    {
+      extends: ["plugin:@typescript-eslint/recommended"],
+      files: ["*.ts", "*.tsx"],
+      parser: "@typescript-eslint/parser",
+      plugins: ["@typescript-eslint"],
+      rules: {
+        "prefer-rest-params": "off",
+        "@typescript-eslint/explicit-module-boundary-types": "off",
+        "@typescript-eslint/no-inferrable-types": "off",
+        "@typescript-eslint/no-explicit-any": "off",
+        "@typescript-eslint/no-this-alias": "off",
+        "@typescript-eslint/consistent-type-imports": "error",
+        "@typescript-eslint/no-unused-vars": [
+          "error",
+          {
+            argsIgnorePattern: "^_",
+            varsIgnorePattern: "^_",
+            ignoreRestSiblings: true,
+            destructuredArrayIgnorePattern: "^_",
+          },
+        ],
+        // This was introduced in 6.0.0
+        "@typescript-eslint/no-unsafe-declaration-merging": "off",
+      },
+    },
+    {
+      extends: [
+        "plugin:jest/recommended",
+        "plugin:jest-dom/recommended",
+        "plugin:testing-library/react",
+      ],
+      plugins: ["jest", "jest-dom", "testing-library"],
+      files: [
+        "*.unit.spec.ts",
+        "*.unit.spec.tsx",
+        "*.unit.spec.js",
+        "*.unit.spec.jsx",
+      ],
+      rules: {
+        "jest/valid-title": ["error", { ignoreTypeOfDescribeName: true }],
+        // In unit tests, you should use `act(() => {})` to invoke things causing state updates
+        // and rerenders to remove lengthy warnings from jest. Disabling the rule allows this.
+        "testing-library/no-unnecessary-act": "off",
```

Comment:
This was removed from master AFAIK, I should rebase this on master and get the latest updates before we merge